src/OpenSpaceToolkit/Astrodynamics/Flight/System/Dynamics/SatelliteDynamics.cpp

```diff
@@ -160,63 +165,47 @@ void SatelliteDynamics::DynamicalEquations(const Dynamics::StateVector& x, Dynam
         totalGravitationalAcceleration_SI += gravitationalAcceleration.inFrame(gcrfSPtr_, currentInstant).getValue();
     }
 
-    // Integrate position and velocity states
+    dxdt[3] += totalGravitationalAcceleration_SI[0];
+    dxdt[4] += totalGravitationalAcceleration_SI[1];
+    dxdt[5] += totalGravitationalAcceleration_SI[2];
+
+    // DRAG
+
+    const Real mass = satelliteSystem_.getMass().inKilograms();
+    const Real dragCoefficient = satelliteSystem_.getDragCoefficient();
+    const Real surfaceArea = satelliteSystem_.getCrossSectionalSurfaceArea();
+
+    for (const auto& objectName : environment_.getObjectNames())
+    {
+        // TBI: currently only defined for Earth
+        if (environment_.accessCelestialObjectWithName(objectName)->accessAtmosphericModel() != nullptr)
+        {
+            const Real atmosphericDensity = environment_.accessCelestialObjectWithName(objectName)
+                                                ->getAtmosphericDensityAt(currentPosition)
+                                                .getValue();
+
+            // [TBI]: Define in Physics celestial body
+            const Vector3d earthAngularVelocity = {0, 0, 7.2921159e-5};  // rad/s
+
+            const Vector3d relativeVelocity =
+                Vector3d(x[3], x[4], x[5]) - earthAngularVelocity.cross(Vector3d(x[0], x[1], x[2]));
+
+            // Calculate drag acceleration
+            const Vector3d dragAcceleration = -(0.5 / mass) * dragCoefficient * surfaceArea * atmosphericDensity *
+                                              relativeVelocity.norm() * relativeVelocity;
+
+            dxdt[3] += dragAcceleration[0];
+            dxdt[4] += dragAcceleration[1];
+            dxdt[5] += dragAcceleration[2];
+        }
+    }
+
+    // Propagate velocity
```

Comment:
Just for clarity for now, do you think it'd be easier to read if we kept dxdt[3, 4, 5] below and compute the 2 contributions we have above and add them at the bottom?

Reply:
Either way works for me. This way has the contributions grouped with their calculations, but the way you're suggesting would make it easier to see every contribution at a glance. Not super impactful IMO but if nobody else comments, I'll just apply this suggestion

Reply:
> would make it easier to see every contribution at a glance
Yeah exactly. This is minor though I agree and it will probably change after we push the dynamics refactor, your call :)